### DISCUSSION TITLE
有关amsthm环境编译问题，以及将theorem和corollary中字体改为italic

### DISCUSSION THREAD

**Discussion by gacktkaga** [sjtug/SJTUThesis#943] [提问]
想请问下如下两个问题。
1.将模版2.0.2中的定理环境从ntheorem改为amsthm时，编译的时似乎会报错，在proof symbol后面会显示“@endpe“，而ntheorem环境下则没有问题，想请问怎么解决？
2. 在新的模版里（语言为en），怎样单独讲theorem和corollary中的直体字体改为italic字体？

**Comment by gacktkaga** [comment]
抱歉，忘记说明我的操作系统时mac Ventura. 第一个问题我找到了如下解决方法：将sjtuthesis.cls中关于amsthm的第1552行的代码“\popQED \endtrivlist \legacy_if_set_false:n { @endpe }”改为 “\popQED\endtrivlist\@endpefalse“，编译就不会报错了。

**Comment by LogCreative** [comment]
使用 `proof` 环境没有遇到类似的错误；由于 `\renewtheorem` 在 `amsthm` 宏包中不存在，这里使用另一种名称来定义新的定理环境。
```latex
\documentclass[lang=en]{sjtuthesis}
\usepackage{amsthm}
\newtheoremstyle{sjtuit}{}{}{\normalfont}{}{\itshape}{}{\ccwd}{}
\theoremstyle{sjtuit}
\newtheorem{thm}{Theorem}[chapter]
\newtheorem{cor}{Corollary}[chapter]
\begin{document}
\begin{proof}
    Test proof
\end{proof}
\begin{theorem}
    Test
\end{theorem}
\begin{thm}
    Test
\end{thm}
\begin{cor}
    Test
\end{cor}
\end{document}
```
